### PR TITLE
Get scan preferences via mqtt

### PR DESCRIPTION
**Pull request by Kraemii**
**What**:
Change openvas to get scan preferences via mqtt instead of redis.
Create main db in openvas, because there is no ospd-openvas connection anymore.

**Why**:

**How**:
Start the eulabeia stack (director, sensor) and start a scan via the example client.

**Checklist**:

- [ ] Tests
- [x] [CHANGELOG](https://github.com/greenbone/openvas/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: src/attack.c [hunk 1]
```diff
@@ -1065,17 +1081,18 @@ attack_network (struct scan_globals *globals)
     connect_main_kb (&alive_hosts_kb);
 
   gettimeofday (&then, NULL);
-
   if (check_kb_access ())
-    return;
-
+    {
+      g_error ("No access to redis kb");
```
Comment (jjnicola) [inline src/attack.c:1086]: Don't use g_error, as suggested in [here](https://developer-old.gnome.org/glib/stable/glib-Message-Logging.html#g-error)
Use g_warning instead, or handle the error with GError.

File: src/attack.c [hunk 2]
```diff
@@ -1065,17 +1081,18 @@ attack_network (struct scan_globals *globals)
...
+      return;
+    }
   /* Init and check Target List */
   hostlist = prefs_get ("TARGET");
   if (hostlist == NULL)
     {
+      g_error ("Target list is empty");
```
Comment (jjnicola) [inline src/attack.c:1093]: Same here.

File: src/attack.c [hunk 3]
```diff
@@ -1085,13 +1102,12 @@ attack_network (struct scan_globals *globals)
         main_kb, "Invalid port list. Ports must be in the range [1-65535]",
         NULL, NULL, "ERRMSG");
       kb_lnk_reset (main_kb);
-      g_warning ("Invalid port list. Ports must be in the range [1-65535]. "
-                 "Scan terminated.");
+      g_error ("Invalid port list. Ports must be in the range [1-65535]. "
```
Comment (jjnicola) [inline src/attack.c:1105]: same here.

File: src/attack.c [hunk 4]
```diff
@@ -1111,19 +1126,19 @@ attack_network (struct scan_globals *globals)
                plugins_init_error);
 
       connect_main_kb (&main_kb);
+      g_error ("%s", buf);
```
Comment (jjnicola) [inline src/attack.c:1129]: same here

File: src/attack.c [hunk 5]
```diff
@@ -1111,19 +1126,19 @@ attack_network (struct scan_globals *globals)
...
       message_to_client (main_kb, buf, NULL, NULL, "ERRMSG");
       kb_lnk_reset (main_kb);
-    }
-
+    };
```
Comment (jjnicola) [inline src/attack.c:1132]: is this `;` necessary ?

File: src/attack.c [hunk 6]
```diff
@@ -1405,8 +1418,14 @@ attack_network (struct scan_globals *globals)
                gvm_hosts_count (hosts));
 
   gvm_hosts_free (hosts);
-  if (test_alive_hosts_only)
-    gvm_hosts_free (alive_hosts_list);
+  // if (test_alive_hosts_only)
+  //   gvm_hosts_free (alive_hosts_list);
 
+  printf ("free maindb\n");
```
Comment (jjnicola) [inline src/attack.c:1424]: Shouldn't be g_message? or better g_debug?

**Review by jjnicola — CHANGES_REQUESTED** [review]

File: src/attack.c [hunk 7]
```diff
@@ -1405,8 +1418,14 @@ attack_network (struct scan_globals *globals)
                gvm_hosts_count (hosts));
 
   gvm_hosts_free (hosts);
-  if (test_alive_hosts_only)
-    gvm_hosts_free (alive_hosts_list);
+  // if (test_alive_hosts_only)
+  //   gvm_hosts_free (alive_hosts_list);
```
[the remark this answers is not in the document]
Comment (Kraemii) [reply src/attack.c:1422]: Ah I wanted to ask something about this, beacause it caused a `free(): invalid pointer` error for some reason. I did not change anything regarding this so I am wondering what causes it.

File: src/attack.c (hunk 6 above)
Comment (Kraemii) [reply src/attack.c:1424]: It was only for debug reasons for my special case. I remove it completely.

File: src/attack.c (hunk 7 above)
[the remark this answers is not in the document]
Comment (Kraemii) [reply src/attack.c:1425]: I left it in and added a `TODO`-comment

**Comment by ArnoStiefvater** [comment]
Closing this PR in favor of https://github.com/greenbone/openvas-scanner/pull/842

**Pull request closed without merging**